src/plot.js

```diff
@@ -106,8 +106,9 @@ export function plot(options = {}) {
     if (
       facetIndex?.length > 1 && // non-trivial faceting
       mark.facet === "auto" && // no explicit mark facet option
+      mark.data &&
       mark.data !== facet.data && // mark not implicitly faceted (different data)
-      (data && data.length) === facetDataLength // mark data seems parallel to facet data
+      arrayify(mark.data).length === facetDataLength // mark data seems parallel to facet data
```

Comment:
This needs to be `arrayify(mark.data)?.length` since mark.data can be nullish… see my previous suggestion.

Reply:
(And then you don’t need the `mark.data` truthy check.)